micro-rdk/src/common/button.rs

```diff
@@ -0,0 +1,84 @@
+use super::{
+    config::ConfigType,
+    generic::DoCommand,
+    status::{Status, StatusError},
+};
+use crate::google;
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+use thiserror::Error;
+
+pub type ButtonType = Arc<Mutex<dyn Button>>;
+
+#[derive(Debug, Error)]
+pub enum ButtonError {
+    #[error("failed to press button")]
+    FailedPress,
+    #[error("test error")]
+    TestError,
```

Comment:
Maybe `OtherButtonError(#[from] Box<dyn std::error::Error + Send + Sync>)`? (see `OtherBoardError` for an example) Then we can define a different error for `FakeButton` and set the precedent for future implementations/modules